internal/core/logic_test.go

```diff
@@ -333,31 +333,66 @@ func TestGetEligibleOperations(t *testing.T) {
 
 	// test conflicts between constraints
 	//
-	// Entirely conflicting constraints (max < min) shall paralyze Castellum and
-	// suppress all actions.
+	// Entirely conflicting constraints of equal priority shall paralyze Castellum and
+	// suppress all actions. Otherwise, the stronger constraint should be enforced.
+	// Priority 0: StrictMinimumSize, StrictMaximumSize
+	// Priority 1: MinimumFreeSize, MinimumSize, MaximumSize
+
 	check(
-		"low=20%, high=80%, crit=95%, step=200%, min=1000",
-		"size=1000, usage=500, smax=900",
-		"", "", // MinimumSize is upheld, but StrictMaximumSize is trying to get us to break it
+		"low=20%, high=80%, crit=95%, step=20%, max=900",
+		"size=1000, usage=500, smin=1100",
+		"high->1200", "high->1100", // StrictMinimumSize takes precedence over maximum constraint
 	)
 	check(
-		"low=20%, high=80%, crit=95%, step=200%, min=1000",
-		"size=900, usage=500, smax=800",
-		"", "", // MinimumSize is already broken, and StrictMaximumSize is trying to get us to break it further
+		"low=20%, high=80%, crit=95%, step=20%, min=900",
+		"size=1000, usage=100, smin=950",
+		"low->950", "low->950", // StrictMinimumSize takes precedence over minimum constraint
 	)
 	check(
-		"low=20%, high=80%, crit=95%, step=200%, max=1000",
+		"low=10%, high=80%, crit=95%, step=20%, min_free=550",
 		"size=1000, usage=500, smin=1100",
-		"", "", // MaximumSize is upheld, but StrictMinimumSize is trying to get us to break it
+		"high->1200", "high->1100", // StrictMinimumSize takes precedence over minimum free constraint
 	)
 	check(
-		"low=20%, high=80%, crit=95%, step=200%, max=1000",
-		"size=1100, usage=500, smin=1200",
-		"", "", // MaximumSize is already broken, and StrictMinimumSize is trying to get us to break it further
+		"low=20%, high=80%, crit=95%, step=20%, min_free=200",
+		"size=1000, usage=900, smax=1050",
+		"high->1050", "high->1050", // StrictMaximumSize takes precedence over minimum free constraint
+	)
+	check(
+		"low=20%, high=80%, crit=95%, step=20%, min_free=200",
+		"size=1000, usage=800, smax=900",
+		"low->900", "low->900", // StrictMaximumSize enforces downsizing over minimum free constraint
+	)
+	check(
+		"low=20%, high=80%, crit=95%, step=20%, min=1100",
+		"size=1000, usage=500, smax=900",
+		"low->800", "low->900", // StrictMaximumSize takes precedence over minimum constraint
```

Comment:
I understand why percentage-step resizing goes to 800 here, but I feel weird about StrictMaximumSize = 900 causing a downsize to 800 here. If the downsize goes through, we will upsize back to 900 immediately afterwards (right?). So I would rather have it go to 900 in both cases.
```suggestion
		"low->900", "low->900", // StrictMaximumSize takes precedence over minimum constraint
```
Having said that, if this turns out to be too difficult to do in this PR without making invasive changes, I'm fine with putting this down as a TODO for later. There are more things to do in the constraint engine soon anyway.

Reply:
In this particular case, there wouldn't be an immediate upsize afterwards, since the new usage would be 62,5% which is within all the thresholds.

`check(
		"low=20%, high=80%, crit=95%, step=20%, min=1100",
		"size=1000, usage=800, smax=900",
		"low->900", "low->900", // StrictMaximumSize takes precedence over minimum constraint
)`
In the case that the forced downsize operation would cross the high threshold, the percentage step strategy will also downsize to 900.

Reply:
Agreed.